3741/1/lib/gem_data_reader.rb

```diff
@@ -0,0 +1,75 @@
+require 'nokogiri'
+require 'json'
+require_relative '../lib/gem_info_decorator'
+
+# parsing data from web page of gem
+class GemDataReader
+  def self.read(gem_name)
+    gem_data_reader = GemDataReader.new(gem_name)
+    gem_data_reader.read
+    gem_data_reader.gem_info
+  end
+
+  attr_reader :gem_name
+  attr_reader :gem_info
+
+  def initialize(gem_name)
+    @gem_name = gem_name
+  end
+
+  def read
+    gem_info = parse_by
```

Comment:
method parse_by doesn't make much sense:
just put the code here and remove parse_by